src/app/core/components/thread/thread.component.ts

```diff
@@ -61,29 +67,71 @@ export class ThreadComponent implements AfterViewInit, OnDestroy {
   readonly itemRoute$ = this.threadService.threadInfo$.pipe(
     map(t => (t ? rawItemRoute('activity', t.itemId) : undefined)),
   );
-
-  private subscription?: Subscription;
+  readonly isThreadStatusOpened$ = this.threadService.threadInfo$.pipe(
+    map(t => (t ? [ 'waiting_for_participant', 'waiting_for_trainer' ].includes(t.status) : false)),
+  );
+  readonly isCurrentUserThreadParticipant$ = this.userCache$.pipe(
+    map(userCache => userCache.some(u => u.isCurrentUser && u.isThreadParticipant)),
+  );
+  allowToOpenThreadState$ = merge(
+    of(readyState(false)),
+    combineLatest([
```

Comment:
I assume the only role of this combineLatest part is just to retrigger the computation when this change. Maybe this should be documented or be done in a more explicit way.

Edit: Ah you also filter on "isThreadOpened" not to do the fetch when the thread is already open... ok that works, but it means the `allowToOpenThreadState$` current value in such a case does not correspond to the current thread, that could be very misleading for future changes.

Reply:
Yes, that part complicated. Need to avoid extra calls to item, need to keep working case when re-open thread without closing thread panel (from threads list), need to trigger call only when open the thread